pallets/runtime/src/asset.rs

```diff
@@ -136,10 +137,53 @@ impl Default for IdentifierType {
     }
 }
 
+/// A wrapper for a token name.
+#[derive(Decode, Encode, Clone, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TokenName(pub Vec<u8>);
+
+impl<T: AsRef<[u8]>> From<T> for TokenName {
+    fn from(s: T) -> Self {
+        let mut v = Vec::new();
```

Comment:
We should use here `Vec::with_capacity(..)` instead of `Vec::new` because we know the size of T.